aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -642,44 +628,128 @@ private CreateSqlCommands getCreateCommands(@NonNull Set<String> modelNames) {
+        for (SQLiteColumn column : table.getSortedColumns()) {
+            final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
+            final Object fieldValue = converter.convertValueFromTarget(model, modelField);
+            bindings.add(fieldValue);
+        }
+        return bindings;
+    }
 
-        // bind model field values to sql columns
-        if (model != null) {
-            final String modelName = getModelName(model);
-            final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
-            final SQLiteModelFieldTypeConverter converter =
-                    new SQLiteModelFieldTypeConverter(schema, modelSchemaRegistry, gson);
-            final Map<String, ModelField> modelFields = schema.getFields();
-
-            final List<SQLiteColumn> columns = sqlCommand.getColumns();
-
-            for (SQLiteColumn column : columns) {
-                final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
-                final Object fieldValue = converter.convertValueFromTarget(model, modelField);
-                bindValueToStatement(compiledSqlStatement, columnIndex, fieldValue);
-                columnIndex++;
+    private List<Model> cascade(Collection<? extends Model> models) {
+        // Use sqliteModelTree to identify every model affected by cascading delete.
+        Map<ModelSchema, Set<String>> descendants =
+                sqliteModelTree.descendantsOf(models);
+
+        List<Model> cascadedModels = new ArrayList<>();
+        for (Map.Entry<ModelSchema, Set<String>> entry : descendants.entrySet()) {
+            ModelSchema schema = entry.getKey();
+            for (String id : entry.getValue()) {
+                // Create dummy model instance using just the ID and model type
+                String dummyJson = gson.toJson(Collections.singletonMap("id", id));
+                Model dummyItem = gson.fromJson(dummyJson, schema.getModelClass());
+                cascadedModels.add(dummyItem);
             }
         }
+        return cascadedModels;
+    }
+
+    private <T extends Model> void writeData(
+            T item,
+            StorageItemChange.Type writeType
+    ) throws DataStoreException {
+        final String modelName = getModelName(item);
+        final ModelSchema modelSchema =
+                modelSchemaRegistry.getModelSchemaForModelClass(modelName);
+        final SQLiteTable sqliteTable = SQLiteTable.fromSchema(modelSchema);
+        final String primaryKeyName = sqliteTable.getPrimaryKeyColumnName();
+        final QueryPredicate matchId = QueryField.field(primaryKeyName).eq(item.getId());
+
+        // Generate SQL command for given action
+        final SqlCommand sqlCommand;
+        final List<Object> bindings;
+        switch (writeType) {
+            case CREATE:
+                LOG.debug("Creating item in " + sqliteTable.getName() +
```

Comment:
This will be logged a lot.  How about making all these verbose?

Reply:
These were already being logged previously haha. I did also have the same concern, so I'll make all of them verbose.